iroh/src/lib.rs

```diff
@@ -233,27 +233,22 @@
 #![deny(missing_docs, rustdoc::broken_intra_doc_links)]
 #![cfg_attr(iroh_docsrs, feature(doc_cfg))]
 
+mod disco;
+mod magicsock;
+
+pub(crate) mod util;
+
 pub mod defaults;
 pub mod dialer;
-mod disco;
 pub mod discovery;
 pub mod dns;
 pub mod endpoint;
-mod magicsock;
 pub mod metrics;
 pub mod protocol;
 pub mod tls;
 
-pub(crate) mod util;
-
-pub use endpoint::{AddrInfo, AddrInfoOptions, Endpoint, NodeAddr, RelayMode};
-pub use iroh_base::{
-    hash, key,
-    key::NodeId,
-    relay_map::{RelayMap, RelayNode, RelayUrl},
-    ticket,
-};
-pub use iroh_relay as relay;
+pub use endpoint::{Endpoint, RelayMode};
+pub use iroh_base::{hash, key, node_addr, relay_map, ticket};
```

Comment:
- do we need `hash`?
- I'd like to keep `NodeId` at least at the toplevel.
- If we're removing `RelayUrl` from here then why keep `RelayMode`?  Yes it's used in the builder but if you want anything else then disabled or default you'll need a whole bunch more relay types anyway.

Reply:
because RelayMode is defined in iroh, not in base

Reply:
> * I'd like to keep `NodeId` at least at the toplevel.

that's fair

Reply:
> * do we need `hash`?

interesting question, we actually don't

Reply:
`hash` is gone now